### PR TITLE
feat: upgrade intl-messageformat

### PR DESCRIPTION
**Pull request by dlockhart**
We'd like to use the latest `intl-messageformat` library in order to leverage version 8's ability to do rich formatting. The release of this PR into BSI will need to be coordinated with [a similar change in `@brightspace-ui/core`](https://github.com/BrightspaceUI/core/pull/3463). No [other BSI things](https://github.com/search?type=code&q=saved%3A%22D2L+Code%22+%22intl-messageformat%22+path%3A%2F%5Epackage.json%2F&saved_searches=%5B%7B%22name%22%3A%22D2L+Code%22%2C%22query%22%3A%22org%3ABrightspace+OR+org%3ABrightspaceUI+OR+org%3ABrightspaceHypermediaComponents+OR+org%3ABrightspaceUILabs%22%7D%5D&expanded_query=org%3ABrightspace+org%3ABrightspaceUI+org%3ABrightspaceHypermediaComponents+org%3ABrightspaceUILabs+%22intl-messageformat%22+path%3A%2F%5Epackage.json%2F) depend on it directly.

Looking through [the CHANGELOG](https://github.com/formatjs/formatjs/blob/main/packages/intl-messageformat/CHANGELOG.md), I don't see any breaking changes that are particularly concerning in versions 8, 9 or 10.

I did a bit of other cleanup while I was in here!

### CODE REVIEW

File: demo/index.html [hunk 1]
```diff
@@ -11,38 +11,10 @@
 	</head>
 	<body unresolved>
 		<d2l-demo-page page-title="d2l-localize-behavior">
-
 			<h2>Behavior (Hybrid)</h2>
-			<div style="margin-bottom: 1rem;">
-				<label>Language:
```
Comment (dlockhart) [inline demo/index.html:17]: No longer needed since the demo page comes with this feature now!

File: .nvmrc [hunk 2]
```diff
@@ -1 +1 @@
-16
\ No newline at end of file
+18
```
Comment (bearfriend) [inline .nvmrc]:
```suggestion
18

```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -17,4 +17,4 @@ jobs:
       - name: Lint
         run: npm run lint
       - name: Unit Tests
-        run: npx web-test-runner --config web-test-runner.config.js --group default --playwright --browsers chromium firefox webkit
+        run: npx web-test-runner --files "./test/**/*.test.js" --node-resolve --playwright --browsers chromium firefox webkit
```
Comment (eKoopmans) [inline .github/workflows/ci.yml:20]: Do we prefer this over a config file?

File: package.json [hunk 4]
```diff
@@ -44,6 +37,6 @@
     "@brightspace-ui/intl": "^3",
     "@polymer/iron-ajax": "^3.0.0-pre.26",
     "@polymer/polymer": "^3",
-    "intl-messageformat": "^7"
+    "intl-messageformat": "^10"
```
Comment (eKoopmans) [inline package.json:40]: To be clear, this is the only line that's actually related to the `intl-messageformat` change - the rest is just cleanup, right?

**Review by eKoopmans — APPROVED** [review]

File: package.json (hunk 4 above)
Comment (dlockhart) [reply package.json:40]: Haha yep.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (dlockhart) [reply .github/workflows/ci.yml:20]: I personally do... haha not sure about "we". For repos like this where the resize-observer thing isn't needed, the extra timeout isn't needed... it seems like overkill to me.

**Review by bearfriend — APPROVED** [review]

**Pull request merged**